Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragmentViewModel.kt

```diff
@@ -208,48 +211,58 @@ class HomeFragmentViewModel @AssistedInject constructor(
         is SubmissionStatePCR.SubmissionDone -> PcrTestSubmissionDoneCard.Item(state)
     }
 
-    private fun RACoronaTest?.toTestCardItem() = when (val state = this.toSubmissionState()) {
-        is SubmissionStateRAT.NoTest -> TestUnregisteredCard.Item(state) {
-            routeToScreen.postValue(HomeFragmentDirections.actionMainFragmentToSubmissionDispatcher())
-        }
-        is SubmissionStateRAT.FetchingResult -> TestFetchingCard.Item(state)
-        is SubmissionStateRAT.TestResultReady -> RapidTestReadyCard.Item(state) {
-            routeToScreen.postValue(
-                HomeFragmentDirections
-                    .actionMainFragmentToSubmissionTestResultAvailableFragment(CoronaTest.Type.RAPID_ANTIGEN)
-            )
-        }
-        is SubmissionStateRAT.TestPositive -> RapidTestPositiveCard.Item(state) {
-            routeToScreen.postValue(
-                HomeFragmentDirections
-                    .actionMainFragmentToSubmissionResultPositiveOtherWarningNoConsentFragment(
-                        CoronaTest.Type.RAPID_ANTIGEN
-                    )
-            )
-        }
-        is SubmissionStateRAT.TestNegative -> RapidTestNegativeCard.Item(state) {
-            routeToScreen.postValue(
-                HomeFragmentDirections
-                    .actionMainFragmentToSubmissionNegativeAntigenTestResultFragment()
-            )
-        }
-        is SubmissionStateRAT.TestInvalid -> RapidTestInvalidCard.Item(state) {
-            popupEvents.postValue(HomeFragmentEvents.ShowDeleteTestDialog)
-        }
-        is SubmissionStateRAT.TestError -> RapidTestErrorCard.Item(state) {
-            routeToScreen.postValue(
-                HomeFragmentDirections
-                    .actionMainFragmentToSubmissionTestResultPendingFragment(testType = CoronaTest.Type.RAPID_ANTIGEN)
-            )
-        }
-        is SubmissionStateRAT.TestPending -> RapidTestPendingCard.Item(state) {
-            routeToScreen.postValue(
-                HomeFragmentDirections
-                    .actionMainFragmentToSubmissionTestResultPendingFragment(testType = CoronaTest.Type.RAPID_ANTIGEN)
-            )
+    private suspend fun coronaTestParameters() = appConfigProvider.getAppConfig().coronaTestParameters
```

Comment:
Please turn this into a `Flow` subscribing to `appCOnfigProvider.latestConfig`, we do not need to recheck the config on UI level via `getAppConfig` the calls happening within other Tasks are sufficient, and if the config changes, we ideally want the card to be updated too. To prevent flickering we could offer the add `distinctUntilChanged`.

Reply:
moved the appconfig resolution to the homeItems flow combine